src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2131,9 +2135,51 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
         _set_remote_or_local_git(cmd, functionapp, resource_group_name, name, deployment_source_url,
                                  deployment_source_branch, deployment_local_git)
 
+    if create_app_insights:
+        try:
+            try_create_application_insights(cmd, functionapp)
+        except Exception:  # pylint: disable=broad-except
+            logger.warning('Error while trying to create and configure an Application Insights for the Function App. '
+                           'Please use the Azure Portal to create and configure the Application Insights, if needed.')
+
     return functionapp
 
 
+def try_create_application_insights(cmd, functionapp):
+    creation_failed_warn = 'Unable to create the Application Insights for the Function App. ' \
+                           'Please use the Azure Portal to manually create and configure the Application Insights, ' \
+                           'if needed.'
+    if not isinstance(functionapp, Site):
```

Comment:
Under what context, the functionapp won't be an instance of `Site`? If it is about creation failure, then Line 2128 would have thrown and failed the whole thing.